builder/buffer.go

```diff
@@ -13,25 +17,74 @@ var escapeCache sync.Map
 
 type escapeCacheKey struct {
 	value  string
-	prefix string
-	suffix string
+	quoter Quoter
 }
 
 // Buffer is used to build query string.
 type Buffer struct {
 	strings.Builder
+	Quoter              Quoter
+	ValueConverter      driver.ValueConverter
 	ArgumentPlaceholder string
 	ArgumentOrdinal     bool
-	EscapePrefix        string
-	EscapeSuffix        string
+	InlineValues        bool
+	BoolTrueValue       string
+	BoolFalseValue      string
 	valueCount          int
 	arguments           []interface{}
 }
 
 // WriteValue query placeholder and append value to argument.
 func (b *Buffer) WriteValue(value interface{}) {
-	b.WritePlaceholder()
-	b.arguments = append(b.arguments, value)
+	if !b.InlineValues {
+		b.WritePlaceholder()
+		b.arguments = append(b.arguments, value)
+		return
+	}
+
+	// Detect float bits to not lose precision after converting to float64
+	var floatBits = 64
+	if reflect.ValueOf(value).Kind() == reflect.Float32 {
```

Comment:
how about using `reflect.Type`? it's less expensive and we can avoid calling `reflect.ValueOf` multiple time